src/transformers/models/bridgetower/modeling_bridgetower.py

```diff
@@ -0,0 +1,1679 @@
+                ]
+            )
+        )
+        self.ln_2 = nn.LayerNorm(config.hidden_size, eps=config.layer_norm_eps)
+        self.attn_mask = None
+
+    def attention(self, hidden_state: torch.Tensor, attention_mask: torch.Tensor):
+        if attention_mask is not None:
+            attention_mask = attention_mask.to(dtype=torch.bool, device=hidden_state.device)
+        self.attn_mask = (
+            self.attn_mask.to(dtype=hidden_state.dtype, device=hidden_state.device)
+            if self.attn_mask is not None
+            else None
+        )
+        return self.attn(
+            hidden_state,
+            hidden_state,
+            hidden_state,
+            need_weights=False,
+            attn_mask=self.attn_mask,
+            key_padding_mask=attention_mask,
+        )[0]
+
+    def forward(self, hidden_state: torch.Tensor, attention_mask: torch.Tensor = None):
+        residual_state = hidden_state + self.attention(self.ln_1(hidden_state), attention_mask)
+        hidden_state = self.ln_2(residual_state)
+        for _, layer in self.mlp.items():
+            hidden_state = layer(hidden_state)
+        hidden_state = residual_state + hidden_state
+        return hidden_state
+
+
+class BridgeTowerTransformer(nn.Module):
+    def __init__(self, config):
+        super().__init__()
+        self.hidden_size = config.hidden_size
+        self.num_hidden_layers = config.num_hidden_layers
+        if config.remove_last_layer:
+            self.resblocks = nn.ModuleList(
+                [BridgeTowerResidualAttention(config) for _ in range(self.num_hidden_layers - 1)]
+            )
+        else:
+            self.resblocks = nn.ModuleList(
+                [BridgeTowerResidualAttention(config) for _ in range(self.num_hidden_layers)]
+            )
+        self.stop_gradient = config.stop_gradient
+
+    def forward(self, hidden_state: torch.Tensor, attention_mask: Optional[torch.Tensor] = None):
+        hidden_states = []
+        for block in self.resblocks:
+            hidden_state = block(hidden_state, attention_mask)
+            if self.stop_gradient:
+                hidden_states.append(hidden_state.detach())
+            else:
+                hidden_states.append(hidden_state)
+        return hidden_states
+
+
+class BridgeTowerVisualTransformer(nn.Module):
+    def __init__(self, config):
+        super().__init__()
+
+        self.conv1 = nn.Conv2d(
+            in_channels=3,
+            out_channels=config.hidden_size,
+            kernel_size=config.patch_size,
+            stride=config.patch_size,
+            bias=False,
+        )
+        scale = config.hidden_size**-0.5
+        self.class_embedding = nn.Parameter(scale * torch.randn(config.hidden_size))
+        self.positional_embedding = nn.Parameter(
+            scale * torch.randn((config.image_size // config.patch_size) ** 2 + 1, config.hidden_size)
+        )
```

Comment:
(nit) usually we wrap these in a `BridgeTowerVisualEmbeddings` class (see the [CLIPVisionTransformer](https://github.com/huggingface/transformers/blob/7119bb052a3f492b9af3afe4f3f13132445eba6e/src/transformers/models/clip/modeling_clip.py#L826) class). It'd be great to make the `BridgeTowerVisualTransformer` class as similar to `CLIPVisionTransformer` in terms of naming and design, leveraging Copied from wherever possible.

Although I'm not sure how much consistency we want to ensure here cc @sgugger

Reply:
We have added a BridgeTowerVisionEmbeddings class. Thanks for the suggestion.